pymatgen/analysis/transition_state.py

```diff
@@ -34,12 +37,47 @@
 __date__ = '6/1/15'
 
 
-class NEBAnalysis(object):
+class NEBAnalysis(MSONable):
     """
     An NEBAnalysis class.
     """
 
-    def __init__(self, outcars, structures, interpolation_order=3):
+    def __init__(self, r, energies, forces, structures, interpolation_order=3):
+        """
+        Initializes an NEBAnalysis from the cumulative root mean squared distances
+        between structures, the energies, the forces, the structures and the
+        interpolation_order for the analysis.
+
+        Args:
+            r: Root mean square distances between structures
+            energies: Energies of each structure along reaction coordinate
+            forces: Tangent forces along the reaction coordinate.
+            structures ([Structure]): List of Structures along reaction
+                coordinate.
+            interpolation_order (int): Order of polynomial to use to
+                interpolate between images. Same format as order parameter in
+                scipy.interplotate.PiecewisePolynomial.
+        """
+        self.r = np.array(r)
+        self.energies = np.array(energies)
+        self.forces = np.array(forces)
+        self.structures = structures
+        self.interpolation_order = interpolation_order
+
+        # We do a piecewise interpolation between the points. Each spline (
+        # cubic by default) is constrained by the boundary conditions of the
+        # energies and the tangent force, i.e., the derivative of
+        # the energy at each pair of points.
+        if scipy_old_piecewisepolynomial:
+            self.spline = PiecewisePolynomial(
+                self.r, np.array([self.energies, -self.forces]).T,
+                orders=interpolation_order)
+        else:
+            # New scipy implementation for scipy > 0.18.0
+            self.spline = CubicSpline(x=self.r, y=self.energies, bc_type=((1, 0.0), (1, 0.0)))
+
+    @classmethod
+    def from_outcars(cls, outcars, structures, interpolation_order=3):
```

Comment:
Pls use **kwargs to passthrough other arguments to __init__. This way you do not need to add arguments every single time a new parameter is added to __init__.